clippy_lints/src/format_impl.rs

```diff
@@ -61,47 +47,93 @@ declare_clippy_lint! {
     "Format trait method called while implementing the same Format trait"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for use of `println`, `print`, `eprintln` or `eprint` in an
+    /// implementation of a formatting trait.
+    ///
+    /// ### Why is this bad?
+    /// Printing during a formatting trait impl is likely unintentional. It can
+    /// cause output to be split between the wrong streams. If `format!` is used
+    /// text would go to stdout/stderr even if not desired.
```

Comment:
I was a little confused, at first I thought you meant using `format!` within the impl. I think you could change the first sentence to `` ...likely unintentional since formatting traits should write to the `Formatter`, not stdout/stderr. `` and stop there.

Reply:
Thanks, changed 👍